Added the ability to return data from a raised intent (#495)
finos/FDC3: docs/api/spec.md

- **Remote API**: An app wants to remote an entire API that it owns to another App.  In this case, the API for the App cannot be standardized.  However, the FDC3 API can address how an App connects to another App in order to get access to a proprietary API.

#### Intents and Context

> When raising an intent a specific context may be provided as input. The type of the provided context may determine which applications can resolve the Intent.

Reviewer: 'intent' has mixed casing in this sentence

> When raising an intent a specific context may be provided as input. The type of the provided context may determine which applications can resolve the Intent.

Author: ```suggestion
When raising an intent a specific context may be provided as input. The type of the provided context may determine which applications can resolve the intent.
```

##### Resolution Object

If the raising of the intent resolves (or rejects), a standard object will be passed into the resolver function with the following format:

```ts
interface IntentResolution {
/**
   * The application that resolved the intent.
   */
  readonly source: TargetApp;
  /**
   * The version number of the Intents schema being used.
   */
  readonly version?: string;

>   /**
>    * Retrieves a promise that will resolve to data returned by the
>    * application that resolves the raised intent. The promise will 
>    * reject if an error is thrown by the intent handler or the promise
>    * returned by the intent handler is reject. If the intent handler 
>    * does not return a promise this function will return null.
>    */
>   getResult(): Promise<Context> | null;

Author: ```suggestion
  /**
   * Retrieves a promise that will resolve to data returned by the
   * application that resolves the raised intent. If an error is thrown by the handler function, 
   * the promise returned is rejected, or no promise is returned then the Desktop Agent 
   * MUST reject the promise returned by the `getResult()` function of the `IntentResolution` 
   * with a string from the `DataError` enumeration.
   */
  getResult(): Promise<Context>;
```

   * chose in response to `fdc3.raiseIntentForContext()`.
   */
  readonly intent: string;
  /**
   * The version number of the Intents schema being used.
   */
  readonly version?: string;

>   /**
>    * Retrieves a promise that will resolve to data returned by the
>    * application that resolves the raised intent. If an error is thrown by the handler function, 
>    * the promise returned is rejected, or no promise is returned then the Desktop Agent 
>    * MUST reject the promise returned by the `getResult()` function of the `IntentResolution` 
>    * with a string from the `DataError` enumeration.
>    */
>   getResult(): Promise<Context>;

Author: ```suggestion
  /**
   * Retrieves a promise that will resolve to data returned by the application that 
   * resolves the raised intent. If an error occurs (i.e. an error is thrown by the handler 
   * function, the promise returned is rejected, or no promise is returned) then the Desktop 
   * Agent MUST reject the promise returned by the `getResult()` function of the 
   * `IntentResolution` with a string from the `DataError` enumeration.
   */
  getResult(): Promise<Context>;
```
